### ISSUE TITLE
Can hang indefinitely after one error from Lichess API

### ISSUE THREAD

**Issue by alex65536** [ShailChoksi/lichess-bot#547]
My installation of lichess-bot failed with the following error:

```
2022-08-15 06:36:15,636 __main__ INFO Challenging a random bot
2022-08-15 06:37:16,914 __main__ ERROR Quitting lichess-bot due to an error:
Traceback (most recent call last):
  File "/opt/sofcheck/lichess-bot/lichess-bot.py", line 923, in <module>
    start_lichess_bot()
  File "/opt/sofcheck/lichess-bot/lichess-bot.py", line 916, in start_lichess_bot
    start(li, user_profile, CONFIG, logging_level, args.logfile)
  File "/opt/sofcheck/lichess-bot/lichess-bot.py", line 271, in start
    matchmaker.challenge()
  File "/opt/sofcheck/lichess-bot/matchmaking.py", line 96, in challenge
    bot_username, base_time, increment, days, variant = self.choose_opponent()
  File "/opt/sofcheck/lichess-bot/matchmaking.py", line 88, in choose_opponent
    online_bots = self.li.get_online_bots()
  File "/opt/sofcheck/lichess-bot/lichess.py", line 142, in get_online_bots
    online_bots = self.api_get(ENDPOINTS["online_bots"], get_raw_text=True)
  File "/opt/sofcheck/venv/lib/python3.9/site-packages/backoff/_sync.py", line 105, in retry
    ret = target(*args, **kwargs)
  File "/opt/sofcheck/lichess-bot/lichess.py", line 69, in api_get
    response.raise_for_status()
  File "/opt/sofcheck/venv/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 502 Server Error: Bad Gateway for url: https://lichess.org/api/bot/online
```

Still, some of the processes worked even after this error, and one of them was consuming 100% CPU.

Possibly, it would be better to retry this kind of temporary errors.

**Comment by MarkZH** [comment]
@alex65536 Can you try the change in PR #549 and let us know if that works for your problem? Thanks.

**Comment by alex65536** [comment]
I may try the proposed PR, but I don't really know how to reproduce it fast. The previous time the bot crashed after running for approximately two weeks.

If you know a simple way way to simulate a faulty Lichess API that yields 502 errors faster, it would be helpful to ensure if the proposed PR really helps.

**Comment by MarkZH** [comment]
What I did was to put `raise HTTPError("Testing")` above `matchmaker.challenge()` on [line 310 in lichess-bot.py](https://github.com/MarkZH/lichess-bot/blob/0d235209d1a8ba55fc942e1b8a8636a916a6f159/lichess-bot.py#L310). This will cause matchmaking to always fail but still allow your bot to accept challenges or play games started manually on lichess.org.

If you haven't already, get the latest version of the PR as I've made a change since your last message.
